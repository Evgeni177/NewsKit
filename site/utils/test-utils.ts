import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme = renderToFragmentWithThemeFactory(
  docsThemeLight,
);
