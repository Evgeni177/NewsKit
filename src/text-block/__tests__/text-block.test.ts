import {renderToFragmentWithTheme} from '../../test/test-utils';
import {TextBlock, TextBlockProps} from '..';

describe('TextBlock', () => {
  test('renders with default styles', () => {
    const fragment = renderToFragmentWithTheme(TextBlock);
    expect(fragment).toMatchSnapshot();
  });

  const tags: Array<['h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'span']> = [
    ['h2'],
    ['span'],
  ];
  test.each(tags)('renders as %s html tag', as => {
    const fragment = renderToFragmentWithTheme(TextBlock, {
      as,
    } as TextBlockProps);
    expect(fragment).toMatchSnapshot();
  });

  test('renders with custom style preset', () => {
    const fragment = renderToFragmentWithTheme(TextBlock, {
      overrides: {
        stylePreset: 'linkPrimary',
      },
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders with custom type preset', () => {
    const fragment = renderToFragmentWithTheme(TextBlock, {
      overrides: {
        typePreset: 'body010',
      },
    });
    expect(fragment).toMatchSnapshot();
  });
});
