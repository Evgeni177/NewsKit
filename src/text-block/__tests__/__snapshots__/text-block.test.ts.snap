// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBlock renders as h2 html tag 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  padding: 1px 0;
  display: block;
}

<h2
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`TextBlock renders as span html tag 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  padding: 1px 0;
}

<span
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`TextBlock renders with custom style preset 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  padding: 1px 0;
  color: #2F75D7;
}

.emotion-0 svg {
  fill: #2F75D7;
  color: #2F75D7;
}

.emotion-0:hover:not(:disabled) {
  color: #2F75D7;
}

.emotion-0:hover:not(:disabled) svg {
  fill: #2F75D7;
  color: #2F75D7;
}

.emotion-0:active:not(:disabled) {
  color: #2F75D7;
}

.emotion-0:active:not(:disabled) svg {
  fill: #2F75D7;
  color: #2F75D7;
}

.emotion-0:visited:not(:disabled) {
  color: #903D91;
}

.emotion-0:visited:not(:disabled) svg {
  fill: #903D91;
  color: #903D91;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`TextBlock renders with custom type preset 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  padding: 1px 0;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 400;
  -webkit-letter-spacing: -0.25px;
  -moz-letter-spacing: -0.25px;
  -ms-letter-spacing: -0.25px;
  letter-spacing: -0.25px;
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.3642857142857143em + 1.2px);
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`TextBlock renders with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  padding: 1px 0;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;
